src/custom/lib/hooks/useCurrencyLogoURIs/useCurrencyLogoURIsMod.ts

```diff
@@ -51,33 +50,43 @@ function getTokenLogoURI(address: string, chainId: SupportedChainId = SupportedC
   }
 }
 
+const currencyLogoCache = new Map<string, Array<string>>()
 // TODO: must be refactored
 export default function useCurrencyLogoURIs(currency?: Currency | null): string[] {
   // There is a modification of Token in useDetectNativeToken()
+  const currencyAddress = currency ? (currency.isNative ? NATIVE_CURRENCY_BUY_ADDRESS : currency.address) : null
```

Comment:
Added caching for results and removed usage of `useHttpLocations()`.
As I can see, we don't have ens name in logo URL, so, `useHttpLocations` looks redundant here.